### PR TITLE
 added streamoupload extractor

### PR DESCRIPTION
**Pull request by Thorodinson1**
[added streamoupload extractor](https://github.com/recloudstream/cloudstream/pull/498) #498

### CODE REVIEW

**Comment by C10udburst** [comment]
you need to add the extractor inside the `utils/ExtractorApi.kt` file

also please see artifact build check as your code has compilation errors

**Comment by Thorodinson1** [comment]
> you need to add the extractor inside the `utils/ExtractorApi.kt` file
> 
> also please see artifact build check as your code has compilation errors

ok

**Comment by Thorodinson1** [comment]
it is still no working i think its because script is obfuscated how can i fix it

**Comment by Thorodinson1** [comment]
> you need to add the extractor inside the `utils/ExtractorApi.kt` file
> 
> also please see artifact build check as your code has compilation errors

it is still no working i think its because script is obfuscated how can i fix it

**Comment by b4byhuey** [comment]
I think your Extractor had same code with SpeedoStream.kt, why dont just put code below into it and StreamoUpload() to utils/ExtractorApi.kt

class StreamoUpload : SpeedoStream() {
	override val name = "StreamoUpload"
    override val mainUrl = "https://streamoupload.xyz"
}

**Comment by Thorodinson1** [comment]
> I think your Extractor had same code with SpeedoStream.kt, why dont just put code below into it and StreamoUpload() to utils/ExtractorApi.kt
> 
> **class StreamoUpload : SpeedoStream() { override val name = "StreamoUpload" override val mainUrl = "https://streamoupload.xyz" }**

it wont work in the code of speedostream script is not obfuscated

**Comment by Thorodinson1** [comment]
what is this error e: /home/runner/work/cloudstream/cloudstream/app/src/main/java/com/lagradost/cloudstream3/utils/ExtractorApi.kt: (402, 5): Unresolved reference: StreamoUpload

e: /home/runner/work/cloudstream/cloudstream/app/src/main/java/com/lagradost/cloudstream3/utils/ExtractorApi.kt: (403, 5): Unresolved reference: StreamoUpload1

**Comment by Thorodinson1** [comment]
> I think your Extractor had same code with SpeedoStream.kt, why dont just put code below into it and StreamoUpload() to utils/ExtractorApi.kt
> 
> **class StreamoUpload : SpeedoStream() { override val name = "StreamoUpload" override val mainUrl = "https://streamoupload.xyz" }**
  script is staring like this    eval(function(p,a,c,k,e,d)  So it won't work

**Comment by Thorodinson1** [comment]
Everything is fixed now let me metre 🥺

File: app/src/main/java/com/lagradost/cloudstream3/extractors/StreamoUpload.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.lagradost.cloudstream3.utils.AppUtils.tryParseJson
+import com.lagradost.cloudstream3.utils.ExtractorApi
+import com.lagradost.cloudstream3.utils.ExtractorLink
+import com.lagradost.cloudstream3.utils.getAndUnpack
+import com.lagradost.cloudstream3.utils.Qualities
+import com.lagradost.cloudstream3.utils.M3u8Helper
+import org.jsoup.select.Elements
+
+class StreamoUpload1 : StreamoUpload() {
+    override val mainUrl = "https://streamoupload.xyz"
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/extractors/StreamoUpload.kt:14]: legit useless class, it overrides the mainurl with the same data so in the end it just creates 2 identical classes. is this a mistake?

**Review by LagradOst — CHANGES_REQUESTED** [review]

**Comment by Thorodinson1** [comment]
fixed the issues

File: app/src/main/java/com/lagradost/cloudstream3/extractors/StreamoUpload.kt (hunk 1 above)
Comment (Thorodinson1) [reply app/src/main/java/com/lagradost/cloudstream3/extractors/StreamoUpload.kt]: done

**Pull request merged**